internal/repository/clickhouse/processed_event.go

```diff
@@ -52,11 +52,8 @@ func (r *ProcessedEventRepository) InsertProcessedEvent(ctx context.Context, eve
 			Mark(ierr.ErrValidation)
 	}
 
-	// Default sign to 1 if not set
+	// Preserve sign as provided (0, 1, or -1)
 	sign := event.Sign
-	if sign == 0 {
-		sign = 1
-	}
```

Comment:
Sign is not intended to be 0 ever. It is supposed to be 1 or -1 
By default it is supposed to be 1 (independent of quantity or anything)
-1 is to enable modifications in future in an auditable way so think of it like:
event 1: quantity = 5 with sign 1

In case it needs to be changed to say 4 then we need to push 2 events
event 2: quantity = 5 with sign -1
event 3: quantity = 4 with sign 1

so total computed will be say sum(quantity * sign) = 5*1 + 5*(-1) + 4*1 = 4